drivers/SmartThings/zwave-sensor/src/homeseer-multi-sensor/init.lua

```diff
@@ -54,11 +56,18 @@ local function update_preferences(self, device, args)
   end
 end
 
-local function device_init(self, device)
-  device:set_update_preferences_fn(update_preferences)
+local function device_refresh(driver, device, args)
```

Comment:
DTH also includes a battery_get in here, I think

Reply:
It does, but it is also being sent here, I'm not entirely sure where it gets sent but according to the tests a get battery does get sent on init.

Reply:
The shared functionality between init and refresh is `update_preferences`, which makes me think that it's the update of the wake interval that's causing a battery report to come in (somehow). I don't know enough about z-wave to know if that makes sense.

> according to the tests a get battery does get sent on init
Only thing that makes me concerned is that `update_preferences` appears to be a no-op if the preference hasn't actually changed. I would say that it makes sense for it to fire on init in the test suite but if somebody triggers a refresh via pull to refresh on a device card then the value won't change so that event might not get emitted. Which in turns means that the pull to refresh won't update the battery.

Also means that we probably need to add another test case since it was pretty easy to construct an adversarial scenario that the existing test wouldn't catch (assuming I'm right which I very well may not be)

Reply:
So I believe the battery check is just built in to some generic wakeup handling and I forgot about it, but doug makes a good point about the update_preferences mechanism that I hadn't considered.

Good to verify with a test of preferences not being changed, but I suspect he is right and we may need to go back to the drawing board.

Reply:
nvm, the battery get is in the battery defaults

we send one when we get a wakeup notification

Reply:
Can confirm after a quick test that without `info_changed` `update_preferences` does appear to be a no-op.

Reply:
> we send one when we get a wakeup notification

So that means this code's good? I need to learn much more about Z-Wave 😅

Reply:
setting the wakeup interval doesn't necessarily trigger a wakeup notification

I think what you may want to do is change the wakeup notification handler to send the refresh command (and also a battery refresh)

the update preferences function will just add itself on to any existing wakeup notification handler